libnetwork/iptables/firewalld.go

```diff
@@ -52,11 +60,49 @@ func firewalldInit() error {
 	if connection, err = newConnection(); err != nil {
 		return fmt.Errorf("Failed to connect to D-Bus system bus: %v", err)
 	}
-	firewalldRunning = checkRunning()
+
+	firewalldRunning, err = checkRunning()
+	if err != nil {
+		return err
+	}
+	// If firewalld is not running on the first check poll
+	if !firewalldRunning {
+		ctx, cancel := context.WithTimeout(context.Background(), fireWalldInitializationTimeout)
+		defer cancel()
+
+		for {
+
+			firewalldRunning, err = func(ctx context.Context) (bool, error) {
+
+				var running bool
+				log.G(context.TODO()).Debugf("Checking status of Firewalld")
+				running, err = checkRunning()
+
+				if err != nil {
+					return running, err
+				}
+				if ctx.Err() != nil {
+					return running, fmt.Errorf("firewalld initialization timed out")
+				}
+				return running, nil
+			}(ctx)
+
+			if firewalldRunning || err != nil {
+				break
+			}
+		}
+
+		if err != nil {
+			log.G(context.TODO()).Debugf("Check running failed: %v", err)
+		}
+	}
+
 	if !firewalldRunning {
-		connection.sysconn.Close()
-		connection = nil
+		log.G(context.TODO()).Debugf("The dbus API is accessible but firewalld is currently unavailable continuing.")
```

Comment:
Continuing? Continuing what?

Reply:
continue initalizing and starting dockerd